lms/models/lti_event.py

```diff
@@ -0,0 +1,94 @@
+from enum import Enum
+from functools import lru_cache
+
+import sqlalchemy as sa
+from sqlalchemy.dialects.postgresql import JSONB
+from sqlalchemy.ext.mutable import MutableDict
+from sqlalchemy import insert
+
+from lms.db import BASE
+
+
+class LTIEventType(BASE):
+    __tablename__ = "lti_event_type"
+
+    class Type(str, Enum):
+        LAUNCH = "launch"
+        DEEP_LINKING = "deep_linking"
+
+    id = sa.Column(sa.Integer, autoincrement=True, primary_key=True)
+    type = sa.Column(
+        "type",
+        sa.Enum(
+            Type,
+            # In order to maintain maximum flexibility we will only enforce the
+            # type on the Python side, and leave the Postgres side open as a plain
+            # VARCHAR
+            native_enum=False,
+            create_constraint=False,
+            validate_strings=True,
+            # Without a length SQLAlchemy will constrain it to the longest value
+            # we happen to have right now, which could limit us in future
+            length=64,
+            # Use the string values, not the keys to persist the values
+            values_callable=lambda obj: [item.value for item in obj],
+        ),
+        nullable=False,
+        unique=False,
+    )
+
+
+class LTIEvent(BASE):
```

Comment:
Explicitly calling this `LTIEvent` and not just event.

The main reason to do this is to continue supporting counting launches with LTI1.3. 

Fixing the new table would involve a fair amount of extra work while here we can just do a one query backfill from the old to the new.

Reply:
I think leaving the old table behind is definitely the way to go.

I'm wondering if the above sentence should read:

> Explicitly calling this LTIEvent and not just _launches_.

Is that what you meant? I wonder by _not_ calling it `Event` what we are trying to rule out? I think there might be events we want to keep track of that aren't strictly speaking anything to do with LTI. 

For example groups reaching a certain level of annotation activity, or users returning after an absence or something.